Tasks/NuGetAuthenticateV0/main.ts

```diff
@@ -0,0 +1,29 @@
+import * as path from 'path';
+import * as tl from 'azure-pipelines-task-lib/task';
+import { installCredProviderToUserProfile, configureCredProvider } from 'artifacts-common/credentialProviderUtils'
+import { ProtocolType } from 'artifacts-common/protocols';
+import { getPackagingServiceConnections } from 'artifacts-common/serviceConnectionUtils'
+import { emitTelemetry } from 'artifacts-common/telemetry'
+
+async function main(): Promise<void> {
+    try {
+        tl.setResourcePath(path.join(__dirname, 'task.json'));
+        tl.setResourcePath(path.join(__dirname, 'node_modules/artifacts-common/module.json'));
+
+        // Install the credential provider
+        const updateCredentialProvider = tl.getBoolInput("updateCredentialProvider", false);
+        await installCredProviderToUserProfile(updateCredentialProvider);
+
+        // Configure the credential provider for both same-organization feeds and service connections
+        const serviceConnections = getPackagingServiceConnections('nuGetServiceConnections');
+        await configureCredProvider(ProtocolType.NuGet, serviceConnections);
+    } catch (error) {
+        tl.setResult(tl.TaskResult.Failed, error);
+    } finally {
+        emitTelemetry("Packaging", "NuGetAuthenticate", {
+            "System.TeamFoundationCollectionUri": tl.getVariable("System.TeamFoundationCollectionUri")
```

Comment:
I'm not emitting anything interesting here yet, and I'm unsure why recording the collection uri is important but we seem to do so in our other tasks.

All the interesting stuff I want to record is in artifacts-common - @jotaylo @shubham90 is there any issue with emitting telemetry multiple times?

Reply:
I don't know why we're logging that everywhere, other than potentially making searches by collection a fraction easier. I'm all for removing it.

There's no issue with emitting telemetry multiple times.